content/en/blog/_posts/2020-12-10-third-party-device-metrics-hits-ga.md

```diff
@@ -0,0 +1,134 @@
+
+Users now need to either install [NVIDIA dcgm-exporter](https://github.com/NVIDIA/gpu-monitoring-tools) or [use bindings](https://github.com/nvidia/go-nvml) to gather more accurate and complete metrics about NVIDIA GPUs and can no longer rely on metrics that were reported by kubelet, such as `nvidia_gpu_duty_cycle` or `nvidia_gpu_memory_used_bytes` which were used to gather NVIDIA GPU memory utilization.
+
+This means that users who used to rely on the NVIDIA GPU metrics reported by kubelet, will need to update their reference and deploy the NVIDIA plugin. Namely the different metrics reported by kubernetes map to the following metrics:
+
+| Kubernetes Metrics              | NVIDIA dcgm-exporter metric                 |
+| ------------------------------- | ------------------------------------------- |
+| `nvidia_gpu_duty_cycle`         | `DCGM_FI_DEV_GPU_UTIL`                      |
+| `nvidia_gpu_memory_used_bytes`  | `DCGM_FI_DEV_FB_USED`                       |
+| `nvidia_gpu_memory_total_bytes` | `DCGM_FI_DEV_FB_FREE + DCGM_FI_DEV_FB_USED` |
+
+You might also be interested in other metrics such as `DCGM_FI_DEV_GPU_TEMP` (the GPU temperature) or DCGM_FI_DEV_POWER_USAGE (the power usage). The [default set](https://github.com/NVIDIA/gpu-monitoring-tools/blob/d5c9bb55b4d1529ca07068b7f81e690921ce2b59/etc/dcgm-exporter/default-counters.csv) is [available here](https://docs.nvidia.com/datacenter/dcgm/latest/dcgm-api/group__dcgmFieldIdentifiers.html) and the full set of metrics of ~300 different metrics is available here.
+
+Note that users keep the ability to set the [DisableAcceleratorMetrics feature gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) to false, effectively re-enabling the ability for kubelet to report NVIDIA GPU metrics.
+
+Paired with the graduation of the Pod Resources API, these tools can be used to generate GPU telemetry [that can be used in visualization dashboards](https://grafana.com/grafana/dashboards/12239), below is an example: 
+
+![Grafana visualization of device metrics](/images/blog/2020-12-10-third-party-device-metrics-hits-ga/grafana.png)
+
+# The Pod Resources API - What can I go on to do with this?
+
+As soon as this interface was introduced, many vendors started using it for widely different use cases! To list a few examples:
+
+The [kuryr-kubernetes](https://github.com/openstack/kuryr-kubernetes) CNI plugin in tandem with [intel-sriov-device-plugin](https://github.com/intel/sriov-network-device-plugin). This allowed the CNI plugin to know which allocation of SR-IOV Virtual Functions (VFs) the kubelet made and use that information to correctly setup the container network namespace and use a device with the appropriate NUMA node. We also expect this interface to be used to track the allocated and available resources with information about the NUMA topology of the worker node.
```

Comment:
> We also expect this interface to be used to track the allocated and available resources with information about the NUMA topology of the worker node.

Should this be a new paragraph?

> The start of this is just listing two projects - are they working together to implement it? or do they both implement it separately?

Kuryr is the CNI plugin while the sriov linked is the device plugin. You can also use the intel sriov plugin though. The possible interactions are complex :) !

> I'd also avoid using 'we also' towards the end, is 'we' the kubernetes project or the two other projects?

We means the "kubernetes maintainers" in that, there are contributors adding features to enable that use case.
Maybe this would be better:

```
The kubernetes community is also working towards enabling this interface to be used to track the allocated and available resources with information about the NUMA topology of the worker node.
```